llama_hub/microsoft_sharepoint/base.py

```diff
@@ -154,9 +162,12 @@ def _get_sharepoint_folder_id(self, folder_path: str) -> str:
         Returns:
             str: The ID of the SharePoint site folder.
         """
-        folder_id_endpoint = (
-            f"{self._drive_id_endpoint}/{self._drive_id}/root:/{folder_path}"
-        )
+        if folder_path == "root":
+            folder_id_endpoint = f"{self._drive_id_endpoint}/{self._drive_id}/root"
```

Comment:
Both if/else can be formatted in the same manner. That is either to use "(" on both or not.

Reply:
I adjusted the way the default argument works to not use `root`, but `""`. Now, the default endpoint is the endpoint that was under `if folder_path == "root"`, and if a `folder_path` is provided (so not ""), then the folder_path is correctly added to the default endpoint. (see latest commit to base)